File: server/src/models/Channel.ts
export interface Channel {
  id: number;
  name: string;
  created_by: string;
  created_at: Date;
  is_active: boolean;
  description?: string;
  color?: string;
}

export interface ChannelMember {
  id: number;
  channel_id: number;
  username: string;
  joined_at: Date;
}

File: app/src/main/java/com/ronin/walkie/model/Channel.kt
package com.ronin.walkie.model

data class Channel(
    val id: Int = 0,
    val name: String = "",
    val description: String = "",
    val color: String = "#4CAF50",
    val created_by: String = "",
    val created_at: String = "",
    val is_active: Boolean = true,
    val member_count: Int = 0
)
